swatch-contracts/src/main/java/com/redhat/swatch/contract/service/ContractService.java

```diff
@@ -170,17 +170,35 @@ public List<Contract> getContracts(
   }
 
   /**
-   * Delete a contract for a given uuid. This is hard delete, because its intended use is for
-   * cleaning up test data.
+   * Delete a contract for a given uuid. This is soft delete. It sets the end date of a contract to
+   * the current timestamp.
    *
    * @param uuid
    */
   @Transactional
   public void deleteContract(String uuid) {
 
-    var isSuccessful = contractRepository.deleteById(UUID.fromString(uuid));
+    updateContractEndDate(uuid, OffsetDateTime.now());
+  }
+
+  /**
+   * Look up a contract via uuid and update the record to have the given end date.
+   *
+   * @param uuid
+   * @param endDate
+   */
+  @Transactional
+  Optional<ContractEntity> updateContractEndDate(String uuid, OffsetDateTime endDate) {
+    var contract = contractRepository.findContract(UUID.fromString(uuid));
+
+    if (Objects.nonNull(contract)) {
```

Comment:
I assuming we should change the end_date only for active contract. In order to check that should you check if the end_date is null?

Reply:
Opting to not do a null check first to allow for flexibility and "weird" testing that QE wants to do